netbox/dcim/api/views.py

```diff
@@ -672,6 +673,12 @@ def list(self, request):
         peer_device_name = request.query_params.get(self._device_param.name)
         peer_interface_name = request.query_params.get(self._interface_param.name)
 
+        # check permissions of user, if login is required
+        if settings.LOGIN_REQUIRED:
```

Comment:
Evaluation of `LOGIN_REQUIRED` is handled by middleware, so this line isn't needed. (We also want to check permissions regardless of how `LOGIN_REQUIRED` is set.)

Reply:
It depends how NetBox is configured. It's simply not necessary to check `LOGIN_REQUIRED` when evaluating permissions.

Reply:
I think that, if the `LOGIN_REQUIRED` should be checked here depends on, if the unauthenticated users should have access or not. It allows them to pass without permission check, when login is not required. Otherwise they are blocked by permission class `IsAuthenticatedOrLoginNotRequired` before they even get here.

If this line is deleted and only 2 following lines are kept, unauthenticated user will always get:
1. `Authentication credentials were not provided` when `LOGIN_REQUIRED = True` (this is ok)
2. `You do not have permission to perform this action` when `LOGIN_REQUIRED = False` (this is the problem)

No way for him to access the resource.

So i think the options are:
1. Users should be always authenticated.
2. `LOGIN_REQUIRED` should be checked here before permissions are checked.
3. I missed something in settings and documentation, that allows setting permissions to unauthenticated users.

If you have other suggestion, please let me know.